solutions/standard/variables.tf

```diff
@@ -0,0 +1,141 @@
+###################################################
+
+variable "mq_s2s_policy_enable" {
+  type        = bool
+  description = "Flag to enable creation of the Service to Service policy to enable the Enterprise Application Service instance to reach MQ instance. Default to false"
+  default     = false
+}
+
+variable "mq_s2s_policy_roles" {
+  type        = list(string)
+  description = "List of roles to be configured for the Service to Service policy. Default to Viewer role."
+  default     = ["Viewer"]
+  validation {
+    condition = alltrue([
+      for role in var.mq_s2s_policy_roles : contains(["Administrator", "Viewer", "Operator", "Editor"], role)
+    ])
+    error_message = "The values of var.mq_s2s_policy_roles must be one or more of \"Administrator\", \"Viewer\", \"Operator\", \"Editor\""
+  }
+}
+
+variable "mq_s2s_policy_source_account_id" {
+  type        = string
+  description = "Source accountID for the Service to Service policy from the Enterprise Application Service instance to MQ instance. If mq_s2s_policy_enable is true and this is left to null the accountID of the API key configured for the provider is used"
+  default     = null
+}
+
+variable "mq_s2s_policy_target_account_id" {
+  type        = string
+  description = "Target accountID for the Service to Service policy from the Enterprise Application Service instance to MQ instance. If mq_s2s_policy_enable is true and this is left to null the accountID of the API key configured for the provider is used"
+  default     = null
+}
+
+variable "mq_s2s_policy_limit_source_resource_flag" {
+  type        = bool
+  description = "Flag to limit the source of the Service to Service policy to the created Enterprise Application Service resource instance ID. If false the Service to Service policy source scope is not limited to the resource instance ID. Configuring scope on the resource instance ID and on the resource group ID is mutually exclusive."
+  default     = true
+}
+
+variable "mq_s2s_policy_limit_source_resource_group_flag" {
+  type        = bool
+  description = "Flag to limit the source of the Service to Service policy to the created Resource Group ID. If false the Service to Service policy source scope is not limited to the resource group ID. Configuring scope on the resource instance ID and on the resource group ID is mutually exclusive."
+  default     = false
+}
+
+variable "mq_s2s_policy_limit_target_resource_id" {
+  type        = string
+  description = "Flag to limit the target of the Service to Service policy to a specific MQ resource instance ID. If null the Service to Service policy target scope is not limited to the resource instance ID"
+  default     = null
+}
+
+variable "mq_s2s_policy_limit_target_resource_group_id" {
+  type        = string
+  description = "Flag to limit the target of the Service to Service policy to a specific resource group ID. If null the Service to Service policy target scope is not limited to the resource group ID"
```

Comment:
The description says "Flag" but I think its the ID you want here for the string input